packages/shared/src/telemetry/clientCache.ts

```diff
@@ -0,0 +1,97 @@
+import type { TelemetryEventRaw } from './types';
+
+type TtlInMilliseconds = number;
+
+const DEFAULT_CACHE_TTL_MS = 1000; // 24 hours
+
+/**
+ * Manages caching for telemetry events using the browser's localStorage to
+ * mitigate event flooding in frequently executed code paths.
+ */
+export class TelemetryClientCache {
```

Comment:
nit: Do you think like `TelemetryEventThrottler`, or something similar, would be a better name? I feel like that more clearly communicates the intent 💭

Reply:
Yeah, it makes sense to me! `ClientCache` is indeed a bit generic, it leads to the question: "caching for what"?